tests/siteAliasTest.php

```diff
@@ -114,4 +114,38 @@ public function testSAList() {
   public function testBadAlias() {
     $this->drush('sa', array('@badalias'), array(), NULL, NULL, self::EXIT_ERROR);
   }
+
+  /**
+   * Ensure that Drush searches deep inside specified search locations
+   * for alias files.
+   */
+  public function testDeepAliasSearching() {
+    $aliasPath = UNISH_SANDBOX . '/aliases';
```

Comment:
No; this is passed as --alias-path, which is always searched recursively, regardless of whether there is a 'site-aliases' directory or not (since the path was explicitly provided by the user).

I could change this to '/site-alias-directory' or something like that, for the avoidance of doubt.